master/internal/api_trials.go

```diff
@@ -158,3 +158,52 @@ func (a *apiServer) KillTrial(
 	}
 	return &resp, err
 }
+
+func (a *apiServer) GetExperimentTrials(
+	_ context.Context, req *apiv1.GetExperimentTrialsRequest,
+) (*apiv1.GetExperimentTrialsResponse, error) {
+	resp := &apiv1.GetExperimentTrialsResponse{}
+
+	switch err := a.m.db.QueryProto(
+		"proto_get_trials_for_experiment",
+		&resp.Trials,
+		req.ExperimentId,
+	); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "experiment %d not found:", req.ExperimentId)
+	case err != nil:
+		return nil, err
+	}
+	a.filter(&resp.Trials, func(i int) bool {
+		v := resp.Trials[i]
+		// If there are no filters, return true.
+		if len(req.States) == 0 {
+			return true
+		}
+
+		// Or if the filters are matched, return true.
+		for _, state := range req.States {
+			if state == v.State {
+				return true
+			}
+		}
+
+		// Else return false.
+		return false
+	})
+
+	a.sort(resp.Trials, req.OrderBy, req.SortBy, apiv1.GetExperimentTrialsRequest_SORT_BY_ID)
+	return resp, a.paginate(&resp.Pagination, &resp.Trials, req.Offset, req.Limit)
+}
+
+func (a *apiServer) GetTrial(_ context.Context, req *apiv1.GetTrialRequest) (
+	resp *apiv1.GetTrialResponse, err error,
+) {
+	switch err := a.m.db.QueryProto("proto_get_trial", &resp.Trial, req.TrialId); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "trial %d not found:", req.TrialId)
+	case err != nil:
+		return nil, err
```

Comment:
nit: use `errors.Wrapf`

Reply:
should we return an error from the status package instead? normal go error would result in a 500 which also seems accurate here.